### ISSUE TITLE
cli doesnt exists on type PostgresConnectionOptions

### ISSUE THREAD

**Issue by carzoglioemanuel** [typeorm/typeorm#8905]
## Issue Description
I cant use the **cli** option with **migrationsDir** in DataSource. Am using version 0.3.6

### Expected Behavior
The documentation say that we can specify the dir where we want to save the new migrations created by cli.

### Actual Behavior
When I use the **cli** option in DataSource, am having an error that say "cli doesnt exists on type PostgresConnectionOptions".
My DataSource:

```
export const AppDataSource = new DataSource({
  type: "postgres",
  host: "localhost",
  port: 5432,
  username: "postgres",
  password: "postgres",
  database: "tesis",
  synchronize: true,
  logging: false,
  entities: [User],
  migrations: [],
  subscribers: [],
  cli: {
    migrationsDir: "src/migration",
  },
});
```

### Steps to Reproduce

Create a new project with typeorm 0.3.6 and initialize the DataSource with **cli** option.

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Windows 10 Pro        |
| Node.js version     | 12.8.1  | 
| Typescript version  | 3.7.5  | 
| TypeORM version     | 0.3.6  | 

### Relevant Database Driver(s)

| DB Type           | Reproducible  |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | no            |
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | yes            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | no            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by muriloux** [comment]
Same here. Also, I tried to create migrations passing the `-d` flag:

`yarn typeorm migration:create -n CreateUser -d src/migration`

It got me this error:

```
$ typeorm-ts-node-commonjs migration:create -n CreateUsers -d src/migration
cli-ts-node-commonjs.js migration:create <path>

Creates a new migration file.

Options:
  -h, --help          Show help                                                                [boolean]
  -o, --outputJs   Generate a migration file on Javascript instead of Typescript
                                                                                   [boolean] [default : false]
  -t, --timestamp  Custom timestamp for the migration name
                                                                                    [número] [default: false]
  -v, --version    Shows the version                              [boolean]

Not enough arguments: Argument 0, need at least 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
I'm using typeorm command in package.json scripts section as it's suggested in the docs.
```
"scripts": {
    ...
    "typeorm": "typeorm-ts-node-commonjs"
}
```
Thanks for any help.

**Comment by muriloux** [comment]
Back here after checking [this issue](https://github.com/typeorm/typeorm/issues/8860).

While no official fixes you can use this workaround suggested by the user Thomas Gorny.

instead of using the `"cli"` option in data-sources.ts or using the `-d ` flag in command line, do something like this:

`yarn typeorm migration:create src/migrations/MigrationName`

No flags, only the path. Apparently the flags aren't working.

**Comment by carzoglioemanuel** [comment]
> Back here after checking [this issue](https://github.com/typeorm/typeorm/issues/8860).
> 
> While no official fixes you can use this workaround suggested by the user Thomas Gorny.
> 
> instead of using the `"cli"` option in data-sources.ts or using the `-d ` flag in command line, do something like this:
> 
> `yarn typeorm migration:create src/migrations/MigrationName`
> 
> No flags, only the path. Apparently the flags aren't working.

Thanks murilo. It works for me.
I hope I don't have another problem related to the cli since I see that some things still need to be fixed ♥

**Comment by pavlovanton** [comment]
Seems documentation is outdated because in the changelog it is mentioned that `cli` options are [removed from the options](https://typeorm.io/changelog#:~:text=new%20CLI%20documentation.-,cli%20option%20from%20BaseConnectionOptions%20(now%20BaseDataSourceOptions%20options)%20was%20removed%20(since%20CLI%20commands%20were%20re%2Dworked),-.)

And this workaround does not work for `migration:show` command

**Comment by muriloux** [comment]
> Seems documentation is outdated because in the changelog it is mentioned that `cli` options are [removed from the options](https://typeorm.io/changelog#:~:text=new%20CLI%20documentation.-,cli%20option%20from%20BaseConnectionOptions%20(now%20BaseDataSourceOptions%20options)%20was%20removed%20(since%20CLI%20commands%20were%20re%2Dworked),-.)
> 
> And this workaround does not work for `migration:show` command

Yes, It turned out I had other problems with cli and decided to downgrade Typeorm to a 0.2 version, since i was just starting a project and refactoring was quite simple.

**Comment by vincevalstro** [comment]
> Back here after checking [this issue](https://github.com/typeorm/typeorm/issues/8860).
> 
> While no official fixes you can use this workaround suggested by the user Thomas Gorny.
> 
> instead of using the `"cli"` option in data-sources.ts or using the `-d ` flag in command line, do something like this:
> 
> `yarn typeorm migration:create src/migrations/MigrationName`
> 
> No flags, only the path. Apparently the flags aren't working.

The `migration:run` does not have the path argument though
-  https://github.com/typeorm/typeorm/issues/8964

**Comment by pezholio** [comment]
If you don't want to keep adding the full path when generating your migrations, then you can add a `migration:generate` command like so:

```
"typeorm:migration:generate": "sh -c 'npm run typeorm -- migration:generate ./path/to/migrations/$0'",
```

**Comment by TGod-Ajayi** [comment]
@pezholio great work-around just one thing though   all migration names keep coming as sh...anyway to set it while running the command ?
